### PR TITLE
Add alpha deployment GitHub Action

### PR DESCRIPTION
**Pull request by carlobeltrame**
- [x] Needs #3043 first!

Copied and reduced / adapted from continuous-deployment.yml

Can be run manually, or will run automatically after the CI on the alpha branch succeeds (when the custom `ci-passed` event from the alpha branch arrives).

Example run in my fork, which only failed because of some missing secrets:
https://github.com/carlobeltrame/ecamp3/actions/runs/3211262214/jobs/5249311690

This can probably be tested on this repository using the workflow_dispatch trigger (manually triggering the workflow on the Actions tab), and selecting the `alpha` branch.

### CODE REVIEW

File: .github/workflows/alpha-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+  upgrade-or-install-deployment:
+    name: Upgrade or install deployment
+    runs-on: ubuntu-latest
+    needs:
+      - select-env-to-deploy
+    if: fromJSON(needs.select-env-to-deploy.outputs.to-deploy)[0] != null
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (BacLuc) [inline .github/workflows/alpha-deployment.yml]: do we need a matrix for that?

File: .github/workflows/alpha-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          helm upgrade --install ecamp3-${{ matrix.deployment.name }} . \
+            --set imageTag=${{ matrix.deployment.sha }} \
+            --set sharedCookieDomain=.ecamp3.ch \
+            --set api.domain=api-${{ matrix.deployment.name }}.ecamp3.ch \
+            --set frontend.domain=${{ matrix.deployment.name }}.ecamp3.ch \
+            --set print.domain=print-${{ matrix.deployment.name }}.ecamp3.ch \
+            --set mail.domain=mail-${{ matrix.deployment.name }}.ecamp3.ch \
+            --set postgresql.enabled=false \
+            --set postgresql.url='${{ secrets.POSTGRES_URL }}/ecamp3${{ matrix.deployment.name }}?sslmode=require' \
+            --set postgresql.adminUrl='${{ secrets.POSTGRES_ADMIN_URL }}/ecamp3${{ matrix.deployment.name }}?sslmode=require' \
```
Comment (BacLuc) [inline .github/workflows/alpha-deployment.yml]:
```suggestion
```

File: .github/workflows/alpha-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            --set postgresql.dropDBOnUninstall=false \
+            --set php.dataMigrationsDir='${{ secrets.DATA_MIGRATIONS_DIR }}' \
+            --set php.appSecret='${{ secrets.API_APP_SECRET }}' \
+            --set php.sentryDsn='${{ secrets.API_SENTRY_DSN }}' \
+            --set php.jwt.passphrase='${{ secrets.JWT_PASSPHRASE }}' \
+            --set php.jwt.publicKey='${{ secrets.JWT_PUBLIC_KEY }}' \
+            --set php.jwt.privateKey='${{ secrets.JWT_PRIVATE_KEY }}' \
+            --set frontend.sentryDsn='${{ secrets.FRONTEND_SENTRY_DSN }}' \
+            --set print.sentryDsn='${{ secrets.PRINT_SENTRY_DSN }}' \
+            --set print.browserWsEndpoint='${{ secrets.BROWSER_WS_ENDPOINT }}' \
```
Comment (BacLuc) [inline .github/workflows/alpha-deployment.yml]:
```suggestion
```

File: .github/workflows/alpha-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+          # later find out which deployments need to be upgraded
+          sed -i 's/^appVersion:.*$/appVersion: "${{ matrix.deployment.sha }}"/' Chart.yaml
+          # Install or upgrade the release
+          helm upgrade --install ecamp3-${{ matrix.deployment.name }} . \
+            --set imageTag=${{ matrix.deployment.sha }} \
+            --set sharedCookieDomain=.ecamp3.ch \
+            --set api.domain=api-${{ matrix.deployment.name }}.ecamp3.ch \
+            --set frontend.domain=${{ matrix.deployment.name }}.ecamp3.ch \
+            --set print.domain=print-${{ matrix.deployment.name }}.ecamp3.ch \
+            --set mail.domain=mail-${{ matrix.deployment.name }}.ecamp3.ch \
```
Comment (BacLuc) [inline .github/workflows/alpha-deployment.yml]:
```suggestion
            --set mail.dsn=${{ secrets.MAILER_DSN }} \
```

File: .github/workflows/alpha-deployment.yml (hunk 1 above)
Comment (carlobeltrame) [reply .github/workflows/alpha-deployment.yml]: Not really. I've simplified the workflow now.

**Review by BacLuc — APPROVED** [review]

**Review by usu — APPROVED** [review]

**Review by pmattmann — APPROVED** [review]

**Pull request merged**